sanowret-crystal.lic

```diff
@@ -0,0 +1,73 @@
+=begin
+	Runs in the background, gazing or exhaling your sanowret crystal. Checks for concentration and current mindstate.
+
+	author: NBSL
+	game: dr
+  version: 0.1 (2017-06-12)
+
+  changelog:
+    0.1 (2017-06-12)
+     * Initial Release
+=end
+
+custom_require.call(%w[common drinfomon])
+
+
+class SanowretCrystal
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'run', regex: /run/i, description: 'Single run of the script', optional: true }
+      ]
+    ]
+    @invalid_room = nil
+    args = parse_args(arg_definitions)
+    check_crystal if args.run
+    passive if args.run.nil?
+  end
+
+  def use_crystal
+    return if @invalid_room == Room.current.id
+    if DRSkill.getxp('Arcana') <= 10
+      reponse = DRC.bput("gaze sanowret crystal", "A soft light blossoms in the very center of the crystal, and begins to fill your mind with the knowledge of \w+.", "However, nothing much else happens, as you lack the concentration to focus.", "This is not a good place for that.")
+    elsif DRSkill.getxp('Arcana') <= 25
+      response = DRC.bput("exhale sanowret crystal", "you come away from the experience with a further understanding of Arcana as the scintillating lights fade again.", "However, nothing much else happens, as you lack the concentration to focus.", "This is not a good place for that.", "Doing that would give away your hiding place.")
+    end
+
+    case response
+    when "Doing that would give away your hiding place."
+      fput("unhide")
+      use_crystal
+    when "This is not a good place for that."
+      @invalid_room = Room.current.id
+      #move XMLData.room_exits.sample
+    end
+  end
+
+  def check_crystal
+    if DRStats.concentration == 100 && DRSkill.getxp('Arcana') <= 25
+      case DRC.bput('tap my sanowret crystal', /^You tap a sanowret crystal inside your .*.$/,/^You tap a sanowret crystal that you are wearing.$/, /^You tap a sanowret crystal that you are holding.$/, "I could not find what you were referring to.")
+      when /You tap a sanowret crystal inside your/
+        fput("get my sanowret crystal")
+        use_crystal
+        fput("stow my sanowret crystal")
+      when /(^You tap a sanowret crystal that you are wearing.$|^You tap a sanowret crystal that you are holding\.$)/
+        use_crystal
+      else
+        echo "Sanowret crystal not found."
+      end
+    end
+  end
+
+  def passive
+    invalid_room = nil
```

Comment:
Ah, the issues editing in 2 places lol. I have it removed in my test one, but forgot to remove it in my forked folder.